### PR TITLE
fix(rolling-upgrade): Set of commits to fix rolling upgrade from 6.0-6.1

### PR DESCRIPTION
**Pull request by aleksbykov**
Set of commits which are fix several issues in rolling upgrades from 6.0 to 6.1

[fix(restore-backup): Restore only existent files](https://github.com/scylladb/scylla-cluster-tests/pull/7525/commits/0fec37f69d339ace09f30355a8917c3029b9c7aa) - Fixing script error when try to restore backup config files which could be missing

[fix(fill_db-UP): Temprorary disable cdc for rolling upgrade](https://github.com/scylladb/scylla-cluster-tests/pull/7525/commits/7cae54dc8d86b12abaf16b6a063a13c9bb8f9e51) - Disable creating table with cdc options because it is not supported with tablets

[fix(upgrade_test): Wait while all nodes up after rollback](https://github.com/scylladb/scylla-cluster-tests/pull/7525/commits/4a923f8705b2e9a2e66a1695dc3982c8b2a0d569) - need to wait after node upgrade/rollback that all nodes are up and normal. it is required by raft topology feature

[fix(get_highest_sstable_version): get enabled sstable version from db](https://github.com/scylladb/scylla-cluster-tests/pull/7525/commits/99d7eebd37bbf467ca86bbc5b56b00590418521a) - Log message with enabled format sstable is dissappeared from log file. Added new methods to get supported sstable format from system_local

[fix(upgrade): no assert on upggrade sstable](https://github.com/scylladb/scylla-cluster-tests/pull/7525/commits/b07f1b9219426e0cf159d87fdde8c1f5ecacba16) - change assert to log error for current sstable format.

Latest to commits could be https://github.com/scylladb/scylladb/issues/18995

### Testing
- [Job with fixes](https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/abykov/job/rolling-upgrade-staging-abykov/job/rolling-upgrade-ubuntu22.04-test/8)

### PR pre-checks (self review)

- [ ] I added the relevant `backport` labels
- [ ] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

### CODE REVIEW

**Comment by aleksbykov** [comment]
Not sure, do we need backport to 6.0

**Comment by fruch** [comment]
> Not sure, do we need backport to 6.0

Take a look at the 6.0 upgrade tests, and you'll know if it's needed or not.

It couldn't hurt does it ?

**Review by fruch — CHANGES_REQUESTED** [review]
@aleksbykov 

the job you mentioned with the fix, didn't reached that part at all

**Comment by soyacz** [comment]
@aleksbykov please mark it as a draft - there are many commented out parts which look like testing only.

**Comment by fruch** [comment]
@aleksbykov it would be helpful if you would communicate a bit of the situation, just marking people as reviewer, doesn't mean it would get attention.

I wasn't even aware this PR is fixing issues that I was assuming tablets team should be fixing(and they didn't as well)

File: test-cases/upgrades/rolling-upgrade.yaml
Comment (yarongilor) [inline test-cases/upgrades/rolling-upgrade.yaml:38]: Is cdc not covered at all? or covered on another upgrade scenario/keyspace?
Do we plan testing both upgrade modes of cdc+vnodes vs. tablets?

Comment (aleksbykov) [reply test-cases/upgrades/rolling-upgrade.yaml:38]: As i know, now it is impossible to enable cdc on table with tablets due to issue #16317. So i disable it temporary for while it not be fixed

Comment (fruch) [reply test-cases/upgrades/rolling-upgrade.yaml:38]: I would expect someone from the team handling tablets to have a list of things needed to be tested with tablets disabled

CDC, and rolling upgrades, or a mix of them, probably need to be on this list.

Comment (yarongilor) [reply test-cases/upgrades/rolling-upgrade.yaml:38]: @fruch , i think the [following doc](https://docs.google.com/document/d/1WNW0Wo27AK9RoPhU38UHmX9TAzeILuoKv8OX4b7kn4Y/edit?usp=sharing) can help with that.

LWT, CDC, Resharding, and few more.
We can either adjust the current longevities for that or cover it under the new folder of ["tablets_disabled" PR](https://github.com/scylladb/scylla-cluster-tests/pull/7655).

**Comment by aleksbykov** [comment]
> @aleksbykov it would be helpful if you would communicate a bit of the situation, just marking people as reviewer, doesn't mean it would get attention.
> 
> I wasn't even aware this PR is fixing issues that I was assuming tablets team should be fixing(and they didn't as well)

I agree my fault, that i didn't noticed. i just wait the fix of https://github.com/scylladb/scylladb/issues/18995 which should be fast, to remove the not need commit, but looks like it is delayed. 
The set of commits  fixes regular rolling upgrades with disabling the cdc.
Could @scylladb/qa-maintainers review the pr?

File: sdcm/fill_db_data.py
Comment (yarongilor) [inline sdcm/fill_db_data.py:3040]: shouldn't it be backported to branch 6.0?

**Review by yarongilor — APPROVED** [review]
LGTM

File: sdcm/fill_db_data.py
Comment (aleksbykov) [reply sdcm/fill_db_data.py:3040]: I didn't found this error upon upgrade from 5.4 to 6.0, looks this issue is only relevant only when create table with cdc only for new tables.

Comment (fruch) [reply sdcm/fill_db_data.py:3040]: Still I think we should backport to 6.0

Next minor release should test 6.0.0 -> 6.0.1, and those changes would be needed

File: test-cases/upgrades/rolling-upgrade.yaml
Comment (fruch) [reply test-cases/upgrades/rolling-upgrade.yaml:38]: @aleksbykov something doesn't add up here, where are you removing cdc from gemini command, and not skipping the gemini command

seems from the code that the gemini command was added for the sake of testing CDC.

now if someone would disable raft and tablets, gemini would still run without cdc.
not sure that the expected outcome.

i.e. one should consider the support matrix of both options, when looking at those changes

File: upgrade_test.py
Comment (fruch) [inline upgrade_test.py:360]: can you elaborate on this change ? 

before we didn't wait for the node to be up normal, before moving to the next one ?

why do we need to check them all ? is the docs for upgrades were update with this addition ?

Comment (fruch) [reply upgrade_test.py:360]: I think we need better references for changing the procedure like that

File: sdcm/fill_db_data.py
Comment (fruch) [reply sdcm/fill_db_data.py:3040]: a suggestion 

we already have `version_cdc_support` which has bunch of relatively old versions

but if one would put the logic to disable cdc related testing into that function.
it would cleanly disable cdc related checks, without doing it in multiple places.

beside this one, that seems to have not be guarded by version_cdc_support

**Review by fruch — COMMENTED** [review]
few suggestions/comments

mainly one should consider the mix of disabling raft topology and tablets, and that those test would still be operational

File: sdcm/fill_db_data.py
Comment (roydahan) [reply sdcm/fill_db_data.py:3040]: @fruch is right, also no one will remember to enable it back.

File: upgrade_test.py
Comment (roydahan) [reply upgrade_test.py:360]: It's safer, we added the same thing in rolling restart.
Before raft, it was a bit faster so we harfly noticed it, but in general of course all should wait for UN.

**Comment by aleksbykov** [comment]
@fruch , can you take a look?
I added new property to check status of tablets. and change a bit gemini part in rolling_upgrade tests. Now if tablets is enabled gemini will be started without cdc and if tablets will be disabled, cdc options will be added to gemini command.
jobs are passed:
 - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/abykov/job/rolling-upgrade-staging-abykov/job/rolling-upgrade-ami-test/20 - Passed
 - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/abykov/job/rolling-upgrade-staging-abykov/job/rolling-upgrade-ubuntu22.04-test/11

I decided not add new sct parameter which could control behavior of enabling/disabling cdc with tablets, and left version_cdc_support method without changes.
i think in next PR this method could be safely removed at all, because we don't have version without cdc support

File: upgrade_test.py
Comment (fruch) [inline upgrade_test.py:683]: #16317  ? in which repo ?

File: upgrade_test.py
Comment (fruch) [inline upgrade_test.py:797]: what is this comment ?

File: sdcm/fill_db_data.py
Comment (fruch) [inline sdcm/fill_db_data.py:3188]: why `optional_cached_property` ? it can return None ?

File: upgrade_test.py
Comment (fruch) [inline upgrade_test.py:491]: I don't understand this code ?

why not replacing this one with the previous ?

File: upgrade_test.py
Comment (fruch) [inline upgrade_test.py:668]: you wrote you gonna remove it in a follow PR, but started removing usage of it here...

File: upgrade_test.py
Comment (fruch) [inline upgrade_test.py:798]: why switching from assert to error print ?

if there's an issue in scylla, please point to it.

and use SkipPerIssue, so it would be stated back when issue is resolved

File: upgrade_test.py
Comment (fruch) [inline upgrade_test.py:685]: I don't understand what's the point of running gemini without cdc ?

if the whole purpose it was introduce was for testing cdc...

Comment (roydahan) [reply upgrade_test.py:685]: It's not the whole point of running Gemini.
Let's not remove it.

File: upgrade_test.py
Comment (aleksbykov) [reply upgrade_test.py]: It was rewritten. repo added

File: upgrade_test.py
Comment (aleksbykov) [reply upgrade_test.py]: This commit was removed. Issue with sstable format was fixed.

File: upgrade_test.py
Comment (aleksbykov) [reply upgrade_test.py]: I renamed the variable to be more meaningful. and also i don't remove old code, which use logs, because  master branch is also used with different version, where old functionality is used.
So if sstable_format was not parsed from logs (as in old versions) then it will be get from scylla table for new versions

File: upgrade_test.py
Comment (aleksbykov) [reply upgrade_test.py:668]: removed this method at all. all latest supported scylla versions supports the cdc.

File: upgrade_test.py
Comment (aleksbykov) [reply upgrade_test.py]: Issue was fixed. i remove the commit at all, because it is not needed anymore

File: upgrade_test.py
Comment (aleksbykov) [reply upgrade_test.py]: a bit change the code but save the main idea, to run gemini during rolling upgrade with and without cdc support

**Comment by aleksbykov** [comment]
new :
 - removed not related commit after issue was fixed: scylladb/scylladb#18995
 - removed old method and vars for version_cdc_support
 - set properities as cached_property
 - add new property to control enabling cdc options for tables based on feature tablets state and SkipPerIssue
 - rename var 'output' -> ''enabled_sstable_format_features' in method 'get_highest_supported_sstable_version' and leave support 2 behaviors get formats from logs and if not found from scylla table

**Comment by aleksbykov** [comment]
@fruch can you take a look?

File: upgrade_test.py
Comment (fruch) [inline upgrade_test.py:488]: I still don't understand this fallback.. and why it's here...

File: upgrade_test.py
Comment (fruch) [reply upgrade_test.py]: reading it from logs doesn't works for new versions  ?
reading it from system tables doesn't work for old versions ?

I'm a bit a lot, what exactly happened that we need to touch any of those parts

if we do, I would want to know why, and make sure we don't keep duplication for no real reason.

**Review by fruch — APPROVED** [review]
LGTM

**Comment by fruch** [comment]
@aleksbykov I would expect a follow up on sstable upgrade part, to clean it clarify it up.

File: upgrade_test.py
Comment (roydahan) [inline upgrade_test.py:1]: @fruch / @syuu1228 IIUC, we need to revert this commit but only from branch-6.1.
In master the tests are from version without jmx to version without jmx.
But in 6.1 we do it from 6.0 that is with jmx (AFAIU) to 6.1 that is without jmx.

Am i right here?

Comment (fruch) [reply upgrade_test.py:1]: what about 2024.2 ? 

I think we just need to fix this change, `s/autobackup/backup/`, and test it to be working

Comment (roydahan) [reply upgrade_test.py:1]: Does 2024.2 have jmx in it or not?

And do we have any documentation or change in procedure for upgrades with jmx and without?

Comment (roydahan) [reply upgrade_test.py:1]: Now @yaronkaikov told me that 6.0 also doesn't have jmx installed.

Comment (roydahan) [reply upgrade_test.py:1]: ah but we didn't have the PR from @aleksbykov to remove it from SCT.

Comment (fruch) [reply upgrade_test.py:1]: Yes, cause the issue is when upgrading from a version that doesn't have jmx

Comment (syuu1228) [reply upgrade_test.py:1]: > @fruch / @syuu1228 IIUC, we need to revert this commit but only from branch-6.1. In master the tests are from version without jmx to version without jmx. But in 6.1 we do it from 6.0 that is with jmx (AFAIU) to 6.1 that is without jmx.

If we won't revert this commit on master, we should apply fix like this:
```
diff --git a/upgrade_test.py b/upgrade_test.py
index 7b8835318..4c9de0425 100644
--- a/upgrade_test.py
+++ b/upgrade_test.py
@@ -117,7 +117,7 @@ def recover_conf(node):
         node.remoter.run(
             r'for conf in $( rpm -qc $(rpm -qa | grep scylla) | grep -v contains ) '
             r'/etc/systemd/system/{var-lib-scylla,var-lib-systemd-coredump}.mount; '
-            r'do if test -e $conf.backup; then sudo cp -v $conf.backup $conf; fi; done')
+            r'do if test -e $conf.autobackup; then sudo cp -v $conf.autobackup $conf; fi; done')
     else:
         node.remoter.run(
             r'for conf in $(cat /var/lib/dpkg/info/scylla-*server.conffiles '
```
Otherwise upgrade test will keep failing.
Since backup_conf() is backuping io.conf to io.conf.autobackup, but recover_conf() is trying to restore the file from io.conf.backup which is not available, and fails.
It is not about the patch purpose (restore only existant files since we dropping java things), it's about backup filename mismatch (maybe mistake on the patch?).

Comment (fruch) [reply upgrade_test.py:1]: Yes it was a mistake in that patch

Since centos8 was deprecated quite before that change was merged, it wasn't noticed until we had centos9 

You have the diff needed please open a PR, and then we could backport it to where it's needed.

Comment (syuu1228) [reply upgrade_test.py:1]: > Yes it was a mistake in that patch
> 
> Since centos8 was deprecated quite before that change was merged, it wasn't noticed until we had centos9
> 
> You have the diff needed please open a PR, and then we could backport it to where it's needed.

Okay I will

**Pull request merged**
